curation/domain_services.py

```diff
@@ -0,0 +1,73 @@
+"""Provide lookup services for functional domains."""
+from pathlib import Path
+from datetime import datetime
+from curation import PROJECT_ROOT
+import logging
+import csv
+from ftplib import FTP
+from typing import List
+
+
+logger = logging.getLogger('fusion_backend')
+logger.setLevel(logging.DEBUG)
+
+
+class DomainService():
+    """Handler class providing requisite services for functional domain lookup."""
+
+    def __init__(self):
+        """Initialize handler class. Download files if necessary, then load and store."""
+        # check if files exist
+        interpro_files = list((PROJECT_ROOT / 'data').glob('interpro_*.tsv'))
+        if len(interpro_files) < 1:
+            print('downloading???')
+            self.download_interpro()
+            interpro_files = list((PROJECT_ROOT / 'data').glob('interpro_*.tsv'))
+        interpro_file: Path = sorted(interpro_files, reverse=True)[0]
+
+        # load file
+        with open(interpro_file) as tsvfile:
+            reader = csv.reader(tsvfile, delimiter='\t')
+            reader.__next__()  # skip header
+            self.domains = {row[2].lower(): {'case': row[2], 'id': row[0]} for row in reader}
+
+    def download_interpro(self):
+        """Retrieve InterPro entry list TSV from EMBL-EBI servers."""
+        logger.info('Downloading InterPro entry list...')
+        today = datetime.today().strftime("%Y%m%d")
+        fpath: Path = PROJECT_ROOT / 'data' / f'interpro_{today}.tsv'
+        try:
+            with FTP('ftp.ebi.ac.uk') as ftp:
+                ftp.login()
+                ftp.cwd('pub/databases/interpro')
+                with open(fpath, 'wb') as fp:
+                    ftp.retrbinary('RETR entry.list', fp.write)
+        except Exception as e:
+            logger.error(f'FTP download failed: {e}')
+            raise Exception(e)
+        logger.info('InterPro entry list download complete.')
+
+    def get_domain_id(self, name: str) -> str:
```

Comment:
Looking at the data file, there are different `ENTRY_TYPE`. Should we only be getting the field if `ENTRY_TYPE = 'Domain'`? Is it useful to store the entry type in the response? I'm unfamiliar with domains, so thought I'd ask.

Reply:
Gonna run this question up the ladder because I also don't know (I think some but not all of them???)